Flaky test TimerStartEventTest.shouldTriggerOnlyTwice

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? it failed once [here](https://github.com/camunda/zeebe/runs/7991555657?check_suite_focus=true) while attempting to merge https://github.com/camunda/zeebe/pull/10153
- Does it block your work? It blocked a merge and may block future ones
- Do we suspect that it is a real failure? No, this seems like a timing issue. The test only has a window of a few seconds. When it's slow (e.g. on CI), it could be unstable.

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Error:  Failures: 
Error:    TimerStartEventTest.shouldTriggerOnlyTwice:1107 
Expecting actual:
  [2251799813685249L, 2251799813685252L, 2251799813685249L, 2251799813685252L]
to contain exactly (and in same order):
  [2251799813685249L,
    2251799813685252L,
    2251799813685249L,
    2251799813685252L,
    2251799813685252L]
but could not find the following elements:
  [2251799813685252L]
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

The test is based on seconds. If the test takes longer than those few seconds it could break.

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

The following logs are from a local reproduction, as the CI didn't show the actual logs.

I was able to reproduce it locally by pinning the Engine clock's time at the start of test and then running the test repeatedly until failure.
```java
engine.getClock().pinCurrentTime();
// ... rest of the test
```

<details><summary>Logs from local reproduction</summary>
<pre>
--------
	[Timestamp] [Partition] ['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  P[partitionId]K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
14:32:02 C DPLY      CREATE            - #001->  -1   -1 - 
14:32:02 E PROC      CREATED           - #002->#001 K001 - process.xml -> "process_1" (version:1)
14:32:02 E TIME      CREATED           - #003->#001 K003 -  @"start_1"[  -1]  in <process "K001"[?]> due T14:32:12.462 reps
14:32:02 E DPLY      CREATED           - #004->#001 K002 - process.xml
14:32:02 E DPLY      FULLY_DISTRIBUTED - #005->#001 K002 - 
14:32:02 C DPLY      CREATE            - #006->  -1   -1 - 
14:32:02 E PROC      CREATED           - #007->#006 K004 - process.xml -> "process_2" (version:1)
14:32:02 E TIME      CREATED           - #008->#006 K006 -  @"start_2"[  -1]  in <process "K004"[?]> due T14:32:12.463 reps
14:32:02 E DPLY      CREATED           - #009->#006 K005 - process.xml
14:32:02 E DPLY      FULLY_DISTRIBUTED - #010->#006 K005 - 
14:32:17 C TIME      TRIGGER           - #011->  -1 K003 -  @"start_1"[  -1]  in <process "K001"[?]> due T14:32:12.462 reps
14:32:17 C TIME      TRIGGER           - #012->  -1 K006 -  @"start_2"[  -1]  in <process "K004"[?]> due T14:32:12.463 reps
14:32:17 E TIME      TRIGGERED         - #013->#011 K003 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:12.462 reps
14:32:17 E PROC_EVNT TRIGGERING        - #014->#011 K008 -  @"start_1"[K001] in <process K001[K007]> (no vars)
14:32:17 C PI        ACTIVATE          - #015->#011 K007 - PROCESS "process_1" in <process "process_1"[K007]>
14:32:17 E TIME      CREATED           - #016->#011 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:22.46
14:32:17 E TIME      TRIGGERED         - #017->#012 K006 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:12.463 reps
14:32:17 E PROC_EVNT TRIGGERING        - #018->#012 K011 -  @"start_2"[K004] in <process K004[K010]> (no vars)
14:32:17 C PI        ACTIVATE          - #019->#012 K010 - PROCESS "process_2" in <process "process_2"[K010]>
14:32:17 E TIME      CREATED           - #020->#012 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:22.462 reps
14:32:17 E PI        ACTIVATING        - #021->#015 K007 - PROCESS "process_1" in <process "process_1"[K007]>
14:32:17 E PI        ACTIVATED         - #022->#015 K007 - PROCESS "process_1" in <process "process_1"[K007]>
14:32:17 E PROC_EVNT TRIGGERED         - #023->#015 K008 -  @"start_1"[K001] in <process K001[K007]> (no vars)
14:32:17 E PI        ACTIVATING        - #024->#015 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
14:32:17 E PI        ACTIVATED         - #025->#015 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
14:32:17 C PI        COMPLETE          - #026->#015 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
14:32:22 E PI        ACTIVATING        - #027->#019 K010 - PROCESS "process_2" in <process "process_2"[K010]>
14:32:22 E PI        ACTIVATED         - #028->#019 K010 - PROCESS "process_2" in <process "process_2"[K010]>
14:32:22 E PROC_EVNT TRIGGERED         - #029->#019 K011 -  @"start_2"[K004] in <process K004[K010]> (no vars)
14:32:22 E PI        ACTIVATING        - #030->#019 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
14:32:22 E PI        ACTIVATED         - #031->#019 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
14:32:22 C PI        COMPLETE          - #032->#019 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
14:32:22 E PI        COMPLETING        - #033->#026 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
14:32:22 E PI        COMPLETED         - #034->#026 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
14:32:22 E PI        SEQ_FLOW_TAKEN    - #035->#026 K015 - SEQUENCE_FLOW "sequenc..030bc05" in <process "process_1"[K007]>
14:32:22 C PI        ACTIVATE          - #036->#026 K016 - END_EVENT "end_1" in <process "process_1"[K007]>
14:32:22 E PI        COMPLETING        - #037->#032 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
14:32:22 E PI        COMPLETED         - #038->#032 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
14:32:22 E PI        SEQ_FLOW_TAKEN    - #039->#032 K017 - SEQUENCE_FLOW "sequenc..c40f048" in <process "process_2"[K010]>
14:32:22 C PI        ACTIVATE          - #040->#032 K018 - END_EVENT "end_2" in <process "process_2"[K010]>
14:32:22 E PI        ACTIVATING        - #041->#036 K016 - END_EVENT "end_1" in <process "process_1"[K007]>
14:32:22 E PI        ACTIVATED         - #042->#036 K016 - END_EVENT "end_1" in <process "process_1"[K007]>
14:32:22 E PI        COMPLETING        - #043->#036 K016 - END_EVENT "end_1" in <process "process_1"[K007]>
14:32:22 E PI        COMPLETED         - #044->#036 K016 - END_EVENT "end_1" in <process "process_1"[K007]>
14:32:22 C PI        COMPLETE          - #045->#036 K007 - PROCESS "process_1" in <process "process_1"[K007]>
14:32:22 C TIME      TRIGGER           - #046->  -1 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:22.46
14:32:22 C TIME      TRIGGER           - #047->  -1 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:22.462 reps
14:32:22 C TIME      TRIGGER           - #048->  -1 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:22.46
14:32:22 C TIME      TRIGGER           - #049->  -1 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:22.462 reps
14:32:22 E PI        ACTIVATING        - #050->#040 K018 - END_EVENT "end_2" in <process "process_2"[K010]>
14:32:22 E PI        ACTIVATED         - #051->#040 K018 - END_EVENT "end_2" in <process "process_2"[K010]>
14:32:22 E PI        COMPLETING        - #052->#040 K018 - END_EVENT "end_2" in <process "process_2"[K010]>
14:32:22 E PI        COMPLETED         - #053->#040 K018 - END_EVENT "end_2" in <process "process_2"[K010]>
14:32:22 C PI        COMPLETE          - #054->#040 K010 - PROCESS "process_2" in <process "process_2"[K010]>
14:32:22 C TIME      TRIGGER           - #055->  -1 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:22.46
14:32:22 C TIME      TRIGGER           - #056->  -1 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:22.462 reps
14:32:22 E PI        COMPLETING        - #057->#045 K007 - PROCESS "process_1" in <process "process_1"[K007]>
14:32:22 E PI        COMPLETED         - #058->#045 K007 - PROCESS "process_1" in <process "process_1"[K007]>
14:32:22 E TIME      TRIGGERED         - #059->#046 K009 -  @"start_1"[  -1]  in <process "K001"[K019]> due T14:32:22.46
14:32:22 E PROC_EVNT TRIGGERING        - #060->#046 K020 -  @"start_1"[K001] in <process K001[K019]> (no vars)
14:32:22 C PI        ACTIVATE          - #061->#046 K019 - PROCESS "process_1" in <process "process_1"[K019]>
14:32:22 E TIME      TRIGGERED         - #062->#047 K012 -  @"start_2"[  -1]  in <process "K004"[K021]> due T14:32:22.462 reps
14:32:22 E PROC_EVNT TRIGGERING        - #063->#047 K022 -  @"start_2"[K004] in <process K004[K021]> (no vars)
14:32:22 C PI        ACTIVATE          - #064->#047 K021 - PROCESS "process_2" in <process "process_2"[K021]>
14:32:22 E TIME      CREATED           - #065->#047 K023 -  @"start_2"[  -1]  in <process "K004"[K021]> due T14:32:32.46
14:32:32 R TIME      TRIGGER           - #066->#048 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:22.46 !NOT_FOUND (Expected to trigger timer with key '2251799813685257', but no such timer was found)
14:32:32 R TIME      TRIGGER           - #067->#049 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:22.462 reps !NOT_FOUND (Expected to trigger timer with key '2251799813685260', but no such timer was found)
14:32:32 E PI        COMPLETING        - #068->#054 K010 - PROCESS "process_2" in <process "process_2"[K010]>
14:32:32 E PI        COMPLETED         - #069->#054 K010 - PROCESS "process_2" in <process "process_2"[K010]>
14:32:32 R TIME      TRIGGER           - #070->#055 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T14:32:22.46 !NOT_FOUND (Expected to trigger timer with key '2251799813685257', but no such timer was found)
14:32:32 R TIME      TRIGGER           - #071->#056 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T14:32:22.462 reps !NOT_FOUND (Expected to trigger timer with key '2251799813685260', but no such timer was found)
14:32:32 E PI        ACTIVATING        - #072->#061 K019 - PROCESS "process_1" in <process "process_1"[K019]>
14:32:32 E PI        ACTIVATED         - #073->#061 K019 - PROCESS "process_1" in <process "process_1"[K019]>
14:32:32 E PROC_EVNT TRIGGERED         - #074->#061 K020 -  @"start_1"[K001] in <process K001[K019]> (no vars)
14:32:32 E PI        ACTIVATING        - #075->#061 K024 - START_EVENT "start_1" in <process "process_1"[K019]>
14:32:32 E PI        ACTIVATED         - #076->#061 K024 - START_EVENT "start_1" in <process "process_1"[K019]>
14:32:32 C PI        COMPLETE          - #077->#061 K024 - START_EVENT "start_1" in <process "process_1"[K019]>
14:32:32 E PI        ACTIVATING        - #078->#064 K021 - PROCESS "process_2" in <process "process_2"[K021]>
14:32:32 E PI        ACTIVATED         - #079->#064 K021 - PROCESS "process_2" in <process "process_2"[K021]>
14:32:32 E PROC_EVNT TRIGGERED         - #080->#064 K022 -  @"start_2"[K004] in <process K004[K021]> (no vars)
14:32:32 E PI        ACTIVATING        - #081->#064 K025 - START_EVENT "start_2" in <process "process_2"[K021]>
14:32:32 E PI        ACTIVATED         - #082->#064 K025 - START_EVENT "start_2" in <process "process_2"[K021]>
14:32:32 C PI        COMPLETE          - #083->#064 K025 - START_EVENT "start_2" in <process "process_2"[K021]>
14:32:32 E PI        COMPLETING        - #084->#077 K024 - START_EVENT "start_1" in <process "process_1"[K019]>
14:32:32 E PI        COMPLETED         - #085->#077 K024 - START_EVENT "start_1" in <process "process_1"[K019]>
14:32:32 E PI        SEQ_FLOW_TAKEN    - #086->#077 K026 - SEQUENCE_FLOW "sequenc..030bc05" in <process "process_1"[K019]>
14:32:32 C PI        ACTIVATE          - #087->#077 K027 - END_EVENT "end_1" in <process "process_1"[K019]>
14:32:32 E PI        COMPLETING        - #088->#083 K025 - START_EVENT "start_2" in <process "process_2"[K021]>
14:32:32 E PI        COMPLETED         - #089->#083 K025 - START_EVENT "start_2" in <process "process_2"[K021]>
14:32:32 E PI        SEQ_FLOW_TAKEN    - #090->#083 K028 - SEQUENCE_FLOW "sequenc..c40f048" in <process "process_2"[K021]>
14:32:32 C PI        ACTIVATE          - #091->#083 K029 - END_EVENT "end_2" in <process "process_2"[K021]>
14:32:32 E PI        ACTIVATING        - #092->#087 K027 - END_EVENT "end_1" in <process "process_1"[K019]>
14:32:32 E PI        ACTIVATED         - #093->#087 K027 - END_EVENT "end_1" in <process "process_1"[K019]>
14:32:32 E PI        COMPLETING        - #094->#087 K027 - END_EVENT "end_1" in <process "process_1"[K019]>
14:32:32 E PI        COMPLETED         - #095->#087 K027 - END_EVENT "end_1" in <process "process_1"[K019]>
14:32:32 C PI        COMPLETE          - #096->#087 K019 - PROCESS "process_1" in <process "process_1"[K019]>
14:32:32 E PI        ACTIVATING        - #097->#091 K029 - END_EVENT "end_2" in <process "process_2"[K021]>
14:32:32 E PI        ACTIVATED         - #098->#091 K029 - END_EVENT "end_2" in <process "process_2"[K021]>
14:32:32 E PI        COMPLETING        - #099->#091 K029 - END_EVENT "end_2" in <process "process_2"[K021]>
14:32:32 E PI        COMPLETED         - #100->#091 K029 - END_EVENT "end_2" in <process "process_2"[K021]>
14:32:32 C PI        COMPLETE          - #101->#091 K021 - PROCESS "process_2" in <process "process_2"[K021]>
14:32:32 E PI        COMPLETING        - #102->#096 K019 - PROCESS "process_1" in <process "process_1"[K019]>
14:32:32 E PI        COMPLETED         - #103->#096 K019 - PROCESS "process_1" in <process "process_1"[K019]>
14:32:32 E PI        COMPLETING        - #104->#101 K021 - PROCESS "process_2" in <process "process_2"[K021]>
14:32:32 E PI        COMPLETED         - #105->#101 K021 - PROCESS "process_2" in <process "process_2"[K021]>

-------------- Deployed Processes ----------------------
process.xml -> "process_1" (version:1)[K001] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_f6f49a32-859b-4747-aec7-43afe2db6879" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process_1" isExecutable="true">
    <startEvent id="start_1" name="start_1">
      <outgoing>sequenceFlow_770fab2d-9719-44ae-b876-c237c030bc05</outgoing>
      <timerEventDefinition id="timerEventDefinition_98dee063-f18a-4cd6-a0a8-2d47ed6741d6">
        <timeCycle id="timeCycle_9e542a25-27e6-4b34-9371-6ca94057862c">R2/2022-08-24T14:32:12.460+02:00[Europe/Berlin]/PT10S</timeCycle>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="end_1" name="end_1">
      <incoming>sequenceFlow_770fab2d-9719-44ae-b876-c237c030bc05</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_770fab2d-9719-44ae-b876-c237c030bc05" sourceRef="start_1" targetRef="end_1"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_a27320fb-42ab-45d6-9d52-e4ad2a6b15fe">
    <bpmndi:BPMNPlane bpmnElement="process_1" id="BPMNPlane_6d93b3aa-6bcd-47f9-b505-3efdd7b9fbdc">
      <bpmndi:BPMNShape bpmnElement="start_1" id="BPMNShape_2494aa78-7348-48f4-b620-4de36cdef259">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="end_1" id="BPMNShape_a00f0699-5202-4195-9749-5e841653a033">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_770fab2d-9719-44ae-b876-c237c030bc05" id="BPMNEdge_785eefc9-b0f4-4042-b417-774dc32909d6">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>

process.xml -> "process_2" (version:1)[K004] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_d937b567-e14d-4778-a5c9-7e092d87d7ea" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process_2" isExecutable="true">
    <startEvent id="start_2" name="start_2">
      <outgoing>sequenceFlow_bb8b06e2-2aea-4b24-a6fe-9eeddc40f048</outgoing>
      <timerEventDefinition id="timerEventDefinition_5df1a4e6-e587-4947-8bbf-740c5379f960">
        <timeCycle id="timeCycle_76231eb7-eca3-4746-a7f4-2c5c7248fe05">R3/2022-08-24T14:32:12.460+02:00[Europe/Berlin]/PT10S</timeCycle>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="end_2" name="end_2">
      <incoming>sequenceFlow_bb8b06e2-2aea-4b24-a6fe-9eeddc40f048</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_bb8b06e2-2aea-4b24-a6fe-9eeddc40f048" sourceRef="start_2" targetRef="end_2"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_ba6b5931-6423-4f75-a9a1-4aa19c81deab">
    <bpmndi:BPMNPlane bpmnElement="process_2" id="BPMNPlane_1e8deaaa-3a1d-4ee3-ac75-2a92713dce26">
      <bpmndi:BPMNShape bpmnElement="start_2" id="BPMNShape_1e1d98f4-da85-4c62-846c-87ff9b20931c">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="end_2" id="BPMNShape_034ad89f-5c4c-4587-9ec9-4cb56171554c">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_bb8b06e2-2aea-4b24-a6fe-9eeddc40f048" id="BPMNEdge_616f4cd5-6e9e-4860-b6f0-874bc8b9ed51">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>


--------------- Decomposed keys (for debugging) -----------------
  -1 <-> -1
K001 <-> 2251799813685249
K002 <-> 2251799813685250
K003 <-> 2251799813685251
K004 <-> 2251799813685252
K005 <-> 2251799813685253
K006 <-> 2251799813685254
K007 <-> 2251799813685255
K008 <-> 2251799813685256
K009 <-> 2251799813685257
K010 <-> 2251799813685258
K011 <-> 2251799813685259
K012 <-> 2251799813685260
K013 <-> 2251799813685261
K014 <-> 2251799813685262
K015 <-> 2251799813685263
K016 <-> 2251799813685264
K017 <-> 2251799813685265
K018 <-> 2251799813685266
K019 <-> 2251799813685267
K020 <-> 2251799813685268
K021 <-> 2251799813685269
K022 <-> 2251799813685270
K023 <-> 2251799813685271
K024 <-> 2251799813685272
K025 <-> 2251799813685273
K026 <-> 2251799813685274
K027 <-> 2251799813685275
K028 <-> 2251799813685276
K029 <-> 2251799813685277

14:32:07.690 [] DEBUG io.camunda.zeebe.broker.logstreams - Close stream processor
...

java.lang.AssertionError: 
Expecting actual:
  [2251799813685249L, 2251799813685252L, 2251799813685249L, 2251799813685252L]
to contain exactly (and in same order):
  [2251799813685249L,
    2251799813685252L,
    2251799813685249L,
    2251799813685252L,
    2251799813685252L]
but could not find the following elements:
  [2251799813685252L]
</pre>
</details>

Not sure but the tests seems to not handle the case that the second model triggers three times:
```

    final BpmnModelInstance firstModel =
        Bpmn.createExecutableProcess("process_1")
            .startEvent("start_1")
            .timerWithCycle(String.format("R2/%s/PT10S", start))
            .endEvent("end_1")
            .done();

    final BpmnModelInstance secondModel =
        Bpmn.createExecutableProcess("process_2")
            .startEvent("start_2")
            .timerWithCycle(String.format("R3/%s/PT10S", start))
            .endEvent("end_2")
            .done();
```

This assert:

```
    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(5))
        .extracting(record -> record.getValue().getProcessDefinitionKey())
        .containsExactly(
            firstDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey,
            firstDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey);
```

fails with:
```
java.lang.AssertionError: 
Expecting actual:
  [2251799813685249L, 2251799813685252L, 2251799813685249L, 2251799813685252L]
to contain exactly (and in same order):
  [2251799813685249L,
    2251799813685252L,
    2251799813685249L,
    2251799813685252L,
    2251799813685252L]
but could not find the following elements:
  [2251799813685252L]


	at io.camunda.zeebe.engine.processing.timer.TimerStartEventTest.shouldTriggerOnlyTwice(TimerStartEventTest.java:1107)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


```

<details>
<summary>Details</summary>
```
13:49:42.842 [Broker-0-LogStream-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
13:49:42.897 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Recovering state of partition 1 from snapshot
13:49:42.897 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@6a070fe5)), function-provider: io.camunda.zeebe.feel.impl.FeelFunctionProvider@64c0fcd5, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@60ff6495, configuration: Configuration(false)]
13:49:42.907 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
13:49:42.908 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@25fdf6ee)), function-provider: io.camunda.zeebe.feel.impl.FeelFunctionProvider@77d7bd93, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@1412d1e8, configuration: Configuration(false)]
13:49:42.909 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@fd2c433)), function-provider: io.camunda.zeebe.feel.impl.FeelFunctionProvider@45d209a1, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@fa31fad, configuration: Configuration(false)]
13:49:42.910 [Broker-0-StreamProcessor-1] INFO  org.camunda.dmn.DmnEngine - DMN-Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@1a85ee25)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@780ed8a8, audit-loggers: List(), configuration: Configuration(false,false,false)]
13:49:42.910 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(org.camunda.dmn.NoUnpackValueMapper@3abfd797)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@780ed8a8, clock: SystemClock, configuration: Configuration(false)]
13:49:42.911 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@441d55b2)), function-provider: io.camunda.zeebe.feel.impl.FeelFunctionProvider@56bcd2ce, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@637deafa, configuration: Configuration(false)]
13:49:42.912 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@27ba8a6d)), function-provider: io.camunda.zeebe.feel.impl.FeelFunctionProvider@9a4ebed, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@393e62b5, configuration: Configuration(false)]
13:49:42.914 [Broker-0-StreamProcessor-1] INFO  org.camunda.dmn.DmnEngine - DMN-Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@1e6f4bc)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@780ed8a8, audit-loggers: List(), configuration: Configuration(false,false,false)]
13:49:42.914 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(org.camunda.dmn.NoUnpackValueMapper@5127ed58)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@780ed8a8, clock: SystemClock, configuration: Configuration(false)]
13:49:42.915 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.processor - Processor starts replay of events. [snapshot-position: -1, replay-mode: PROCESSING]
13:49:42.916 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.processor - Processor finished replay, with [lastProcessedPosition: -1, lastWrittenPosition: -1]
13:49:42.916 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Starting processing of migration tasks (use LogLevel.DEBUG for more details) ... 
13:49:42.916 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.engine.state.migration - Found 3 migration tasks: ProcessMessageSubscriptionSentTimeMigration, MessageSubscriptionSentTimeMigration, TemporaryVariableMigration
13:49:42.916 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Skipping ProcessMessageSubscriptionSentTimeMigration migration (1/3).  It was determined it does not need to run right now.
13:49:42.916 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Skipping MessageSubscriptionSentTimeMigration migration (2/3).  It was determined it does not need to run right now.
13:49:42.917 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Skipping TemporaryVariableMigration migration (3/3).  It was determined it does not need to run right now.
13:49:42.917 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Completed processing of migration tasks (use LogLevel.DEBUG for more details) ... 
13:49:42.917 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.engine.state.migration - Executed 0 migration tasks: 
13:49:48.073 [] INFO  io.camunda.zeebe.test - Test failed, following records were exported:
13:49:48.248 [] INFO  io.camunda.zeebe.test - Compact log representation:
--------
	[Timestamp] ['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position] K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
13:49:42 C DPLY      CREATE            - #001->  -1   -1 - 
13:49:42 E PROC      CREATED           - #002->#001 K001 - process.xml -> "process_1" (version:1)
13:49:42 E TIME      CREATED           - #003->#001 K003 -  @"start_1"[  -1]  in <process "K001"[?]> due T13:49:52.9172 reps
13:49:42 E DPLY      CREATED           - #004->#001 K002 - process.xml
13:49:42 E DPLY      FULLY_DISTRIBUTED - #005->#001 K002 - 
13:49:42 C DPLY      CREATE            - #006->  -1   -1 - 
13:49:42 E PROC      CREATED           - #007->#006 K004 - process.xml -> "process_2" (version:1)
13:49:42 E TIME      CREATED           - #008->#006 K006 -  @"start_2"[  -1]  in <process "K004"[?]> due T13:49:52.9173 reps
13:49:42 E DPLY      CREATED           - #009->#006 K005 - process.xml
13:49:42 E DPLY      FULLY_DISTRIBUTED - #010->#006 K005 - 
13:49:58 C TIME      TRIGGER           - #011->  -1 K003 -  @"start_1"[  -1]  in <process "K001"[?]> due T13:49:52.9172 reps
13:49:58 C TIME      TRIGGER           - #012->  -1 K006 -  @"start_2"[  -1]  in <process "K004"[?]> due T13:49:52.9173 reps
13:49:58 E TIME      TRIGGERED         - #013->#011 K003 -  @"start_1"[  -1]  in <process "K001"[K007]> due T13:49:52.9172 reps
13:49:58 E PROC_EVNT TRIGGERING        - #014->#011 K008 -  @"start_1"[K001] in <process K001[K007]> (no vars)
13:49:58 C PI        ACTIVATE          - #015->#011 K007 - PROCESS "process_1" in <process "process_1"[K007]>
13:49:58 E TIME      CREATED           - #016->#011 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T13:50:02.917
13:49:58 C TIME      TRIGGER           - #017->  -1 K006 -  @"start_2"[  -1]  in <process "K004"[?]> due T13:49:52.9173 reps
13:49:58 E TIME      TRIGGERED         - #018->#012 K006 -  @"start_2"[  -1]  in <process "K004"[K010]> due T13:49:52.9173 reps
13:49:58 E PROC_EVNT TRIGGERING        - #019->#012 K011 -  @"start_2"[K004] in <process K004[K010]> (no vars)
13:49:58 C PI        ACTIVATE          - #020->#012 K010 - PROCESS "process_2" in <process "process_2"[K010]>
13:49:58 E TIME      CREATED           - #021->#012 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T13:50:02.9172 reps
13:50:03 C TIME      TRIGGER           - #022->  -1 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T13:50:02.917
13:50:03 C TIME      TRIGGER           - #023->  -1 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T13:50:02.9172 reps
13:50:03 E PI        ACTIVATING        - #024->#015 K007 - PROCESS "process_1" in <process "process_1"[K007]>
13:50:03 E PI        ACTIVATED         - #025->#015 K007 - PROCESS "process_1" in <process "process_1"[K007]>
13:50:03 E PROC_EVNT TRIGGERED         - #026->#015 K008 -  @"start_1"[K001] in <process K001[K007]> (no vars)
13:50:03 E PI        ACTIVATING        - #027->#015 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
13:50:03 E PI        ACTIVATED         - #028->#015 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
13:50:03 C PI        COMPLETE          - #029->#015 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
13:50:03 R TIME      TRIGGER           - #030->#017 K006 -  @"start_2"[  -1]  in <process "K004"[?]> due T13:49:52.9173 reps !NOT_FOUND (Expected to trigger timer with key '2251799813685254', but no such timer was found)
13:50:03 C TIME      TRIGGER           - #031->  -1 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T13:50:02.917
13:50:03 C TIME      TRIGGER           - #032->  -1 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T13:50:02.9172 reps
13:50:03 E PI        ACTIVATING        - #033->#020 K010 - PROCESS "process_2" in <process "process_2"[K010]>
13:50:03 E PI        ACTIVATED         - #034->#020 K010 - PROCESS "process_2" in <process "process_2"[K010]>
13:50:03 E PROC_EVNT TRIGGERED         - #035->#020 K011 -  @"start_2"[K004] in <process K004[K010]> (no vars)
13:50:03 E PI        ACTIVATING        - #036->#020 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
13:50:03 E PI        ACTIVATED         - #037->#020 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
13:50:03 C PI        COMPLETE          - #038->#020 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
13:50:03 E TIME      TRIGGERED         - #039->#022 K009 -  @"start_1"[  -1]  in <process "K001"[K015]> due T13:50:02.917
13:50:03 E PROC_EVNT TRIGGERING        - #040->#022 K016 -  @"start_1"[K001] in <process K001[K015]> (no vars)
13:50:03 C PI        ACTIVATE          - #041->#022 K015 - PROCESS "process_1" in <process "process_1"[K015]>
13:50:03 E TIME      TRIGGERED         - #042->#023 K012 -  @"start_2"[  -1]  in <process "K004"[K017]> due T13:50:02.9172 reps
13:50:03 E PROC_EVNT TRIGGERING        - #043->#023 K018 -  @"start_2"[K004] in <process K004[K017]> (no vars)
13:50:03 C PI        ACTIVATE          - #044->#023 K017 - PROCESS "process_2" in <process "process_2"[K017]>
13:50:03 E TIME      CREATED           - #045->#023 K019 -  @"start_2"[  -1]  in <process "K004"[K017]> due T13:50:12.917
13:50:13 E PI        COMPLETING        - #046->#029 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
13:50:13 E PI        COMPLETED         - #047->#029 K013 - START_EVENT "start_1" in <process "process_1"[K007]>
13:50:13 E PI        SEQ_FLOW_TAKEN    - #048->#029 K020 - SEQUENCE_FLOW "sequenc..549f327" in <process "process_1"[K007]>
13:50:13 C PI        ACTIVATE          - #049->#029 K021 - END_EVENT "end_1" in <process "process_1"[K007]>
13:50:13 R TIME      TRIGGER           - #050->#031 K009 -  @"start_1"[  -1]  in <process "K001"[K007]> due T13:50:02.917 !NOT_FOUND (Expected to trigger timer with key '2251799813685257', but no such timer was found)
13:50:13 R TIME      TRIGGER           - #051->#032 K012 -  @"start_2"[  -1]  in <process "K004"[K010]> due T13:50:02.9172 reps !NOT_FOUND (Expected to trigger timer with key '2251799813685260', but no such timer was found)
13:50:13 E PI        COMPLETING        - #052->#038 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
13:50:13 E PI        COMPLETED         - #053->#038 K014 - START_EVENT "start_2" in <process "process_2"[K010]>
13:50:13 E PI        SEQ_FLOW_TAKEN    - #054->#038 K022 - SEQUENCE_FLOW "sequenc..4c33552" in <process "process_2"[K010]>
13:50:13 C PI        ACTIVATE          - #055->#038 K023 - END_EVENT "end_2" in <process "process_2"[K010]>
13:50:13 E PI        ACTIVATING        - #056->#041 K015 - PROCESS "process_1" in <process "process_1"[K015]>
13:50:13 E PI        ACTIVATED         - #057->#041 K015 - PROCESS "process_1" in <process "process_1"[K015]>
13:50:13 E PROC_EVNT TRIGGERED         - #058->#041 K016 -  @"start_1"[K001] in <process K001[K015]> (no vars)
13:50:13 E PI        ACTIVATING        - #059->#041 K024 - START_EVENT "start_1" in <process "process_1"[K015]>
13:50:13 E PI        ACTIVATED         - #060->#041 K024 - START_EVENT "start_1" in <process "process_1"[K015]>
13:50:13 C PI        COMPLETE          - #061->#041 K024 - START_EVENT "start_1" in <process "process_1"[K015]>
13:50:13 E PI        ACTIVATING        - #062->#044 K017 - PROCESS "process_2" in <process "process_2"[K017]>
13:50:13 E PI        ACTIVATED         - #063->#044 K017 - PROCESS "process_2" in <process "process_2"[K017]>
13:50:13 E PROC_EVNT TRIGGERED         - #064->#044 K018 -  @"start_2"[K004] in <process K004[K017]> (no vars)
13:50:13 E PI        ACTIVATING        - #065->#044 K025 - START_EVENT "start_2" in <process "process_2"[K017]>
13:50:13 E PI        ACTIVATED         - #066->#044 K025 - START_EVENT "start_2" in <process "process_2"[K017]>
13:50:13 C PI        COMPLETE          - #067->#044 K025 - START_EVENT "start_2" in <process "process_2"[K017]>
13:50:13 E PI        ACTIVATING        - #068->#049 K021 - END_EVENT "end_1" in <process "process_1"[K007]>
13:50:13 E PI        ACTIVATED         - #069->#049 K021 - END_EVENT "end_1" in <process "process_1"[K007]>
13:50:13 E PI        COMPLETING        - #070->#049 K021 - END_EVENT "end_1" in <process "process_1"[K007]>
13:50:13 E PI        COMPLETED         - #071->#049 K021 - END_EVENT "end_1" in <process "process_1"[K007]>
13:50:13 C PI        COMPLETE          - #072->#049 K007 - PROCESS "process_1" in <process "process_1"[K007]>
13:50:13 E PI        ACTIVATING        - #073->#055 K023 - END_EVENT "end_2" in <process "process_2"[K010]>
13:50:13 E PI        ACTIVATED         - #074->#055 K023 - END_EVENT "end_2" in <process "process_2"[K010]>
13:50:13 E PI        COMPLETING        - #075->#055 K023 - END_EVENT "end_2" in <process "process_2"[K010]>
13:50:13 E PI        COMPLETED         - #076->#055 K023 - END_EVENT "end_2" in <process "process_2"[K010]>
13:50:13 C PI        COMPLETE          - #077->#055 K010 - PROCESS "process_2" in <process "process_2"[K010]>
13:50:13 E PI        COMPLETING        - #078->#061 K024 - START_EVENT "start_1" in <process "process_1"[K015]>
13:50:13 E PI        COMPLETED         - #079->#061 K024 - START_EVENT "start_1" in <process "process_1"[K015]>
13:50:13 E PI        SEQ_FLOW_TAKEN    - #080->#061 K026 - SEQUENCE_FLOW "sequenc..549f327" in <process "process_1"[K015]>
13:50:13 C PI        ACTIVATE          - #081->#061 K027 - END_EVENT "end_1" in <process "process_1"[K015]>
13:50:13 E PI        COMPLETING        - #082->#067 K025 - START_EVENT "start_2" in <process "process_2"[K017]>
13:50:13 E PI        COMPLETED         - #083->#067 K025 - START_EVENT "start_2" in <process "process_2"[K017]>
13:50:13 E PI        SEQ_FLOW_TAKEN    - #084->#067 K028 - SEQUENCE_FLOW "sequenc..4c33552" in <process "process_2"[K017]>
13:50:13 C PI        ACTIVATE          - #085->#067 K029 - END_EVENT "end_2" in <process "process_2"[K017]>
13:50:13 E PI        COMPLETING        - #086->#072 K007 - PROCESS "process_1" in <process "process_1"[K007]>
13:50:13 E PI        COMPLETED         - #087->#072 K007 - PROCESS "process_1" in <process "process_1"[K007]>
13:50:13 E PI        COMPLETING        - #088->#077 K010 - PROCESS "process_2" in <process "process_2"[K010]>
13:50:13 E PI        COMPLETED         - #089->#077 K010 - PROCESS "process_2" in <process "process_2"[K010]>
13:50:13 E PI        ACTIVATING        - #090->#081 K027 - END_EVENT "end_1" in <process "process_1"[K015]>
13:50:13 E PI        ACTIVATED         - #091->#081 K027 - END_EVENT "end_1" in <process "process_1"[K015]>
13:50:13 E PI        COMPLETING        - #092->#081 K027 - END_EVENT "end_1" in <process "process_1"[K015]>
13:50:13 E PI        COMPLETED         - #093->#081 K027 - END_EVENT "end_1" in <process "process_1"[K015]>
13:50:13 C PI        COMPLETE          - #094->#081 K015 - PROCESS "process_1" in <process "process_1"[K015]>
13:50:13 E PI        ACTIVATING        - #095->#085 K029 - END_EVENT "end_2" in <process "process_2"[K017]>
13:50:13 E PI        ACTIVATED         - #096->#085 K029 - END_EVENT "end_2" in <process "process_2"[K017]>
13:50:13 E PI        COMPLETING        - #097->#085 K029 - END_EVENT "end_2" in <process "process_2"[K017]>
13:50:13 E PI        COMPLETED         - #098->#085 K029 - END_EVENT "end_2" in <process "process_2"[K017]>
13:50:13 C PI        COMPLETE          - #099->#085 K017 - PROCESS "process_2" in <process "process_2"[K017]>
13:50:13 E PI        COMPLETING        - #100->#094 K015 - PROCESS "process_1" in <process "process_1"[K015]>
13:50:13 E PI        COMPLETED         - #101->#094 K015 - PROCESS "process_1" in <process "process_1"[K015]>
13:50:13 E PI        COMPLETING        - #102->#099 K017 - PROCESS "process_2" in <process "process_2"[K017]>
13:50:13 E PI        COMPLETED         - #103->#099 K017 - PROCESS "process_2" in <process "process_2"[K017]>

-------------- Deployed Processes ----------------------
process.xml -> "process_1" (version:1)[K001] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_fb0d2381-e5b7-4b6f-b4c1-8a779dd076f1" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process_1" isExecutable="true">
    <startEvent id="start_1" name="start_1">
      <outgoing>sequenceFlow_1c206563-d354-4448-bce6-1c05b549f327</outgoing>
      <timerEventDefinition id="timerEventDefinition_672f5189-e09c-4e62-a5d0-335b5dbc58ea">
        <timeCycle id="timeCycle_d1e0d5d6-8b1d-4a95-ad80-5528f6ffcec7">R2/2022-08-25T13:49:52.917+02:00[Europe/Berlin]/PT10S</timeCycle>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="end_1" name="end_1">
      <incoming>sequenceFlow_1c206563-d354-4448-bce6-1c05b549f327</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_1c206563-d354-4448-bce6-1c05b549f327" sourceRef="start_1" targetRef="end_1"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_68632332-5bfa-4b25-96bb-9872a6c368b0">
    <bpmndi:BPMNPlane bpmnElement="process_1" id="BPMNPlane_ab07482d-b98d-41d7-9df2-15e01c2277c3">
      <bpmndi:BPMNShape bpmnElement="start_1" id="BPMNShape_dee42889-dc3b-40b3-bb7e-dee89d1b8bb4">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="end_1" id="BPMNShape_ad74ae6a-2a27-45b5-a587-fe927b18fec9">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_1c206563-d354-4448-bce6-1c05b549f327" id="BPMNEdge_bc4ede0d-d7df-474d-9cd4-72852891e96c">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>

process.xml -> "process_2" (version:1)[K004] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_2c1afeed-6fb6-42ec-8289-b3ae299dce25" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process_2" isExecutable="true">
    <startEvent id="start_2" name="start_2">
      <outgoing>sequenceFlow_fde20e38-fe7a-4be3-818a-d11524c33552</outgoing>
      <timerEventDefinition id="timerEventDefinition_309e055a-d396-41e2-be25-c85695cbb941">
        <timeCycle id="timeCycle_4b36db46-a7c4-4b61-97bf-7ac14d106016">R3/2022-08-25T13:49:52.917+02:00[Europe/Berlin]/PT10S</timeCycle>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="end_2" name="end_2">
      <incoming>sequenceFlow_fde20e38-fe7a-4be3-818a-d11524c33552</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_fde20e38-fe7a-4be3-818a-d11524c33552" sourceRef="start_2" targetRef="end_2"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_7a143f55-9acf-45bb-9c32-c27b609aad9c">
    <bpmndi:BPMNPlane bpmnElement="process_2" id="BPMNPlane_bf45a3fc-315c-482d-86d9-92dc306ec8e4">
      <bpmndi:BPMNShape bpmnElement="start_2" id="BPMNShape_eb1dbd63-4589-4d42-9411-59657c0e4a92">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="end_2" id="BPMNShape_2cfb55dc-9424-4cab-a3e7-f93051425b7b">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_fde20e38-fe7a-4be3-818a-d11524c33552" id="BPMNEdge_b2003062-940e-4459-a211-e4fc6a0035d6">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>


--------------- Decomposed keys (for debugging) -----------------
  -1 <-> -1
K001 <-> 2251799813685249
K002 <-> 2251799813685250
K003 <-> 2251799813685251
K004 <-> 2251799813685252
K005 <-> 2251799813685253
K006 <-> 2251799813685254
K007 <-> 2251799813685255
K008 <-> 2251799813685256
K009 <-> 2251799813685257
K010 <-> 2251799813685258
K011 <-> 2251799813685259
K012 <-> 2251799813685260
K013 <-> 2251799813685261
K014 <-> 2251799813685262
K015 <-> 2251799813685263
K016 <-> 2251799813685264
K017 <-> 2251799813685265
K018 <-> 2251799813685266
K019 <-> 2251799813685267
K020 <-> 2251799813685268
K021 <-> 2251799813685269
K022 <-> 2251799813685270
K023 <-> 2251799813685271
K024 <-> 2251799813685272
K025 <-> 2251799813685273
K026 <-> 2251799813685274
K027 <-> 2251799813685275
K028 <-> 2251799813685276
K029 <-> 2251799813685277

13:49:48.250 [] DEBUG io.camunda.zeebe.broker.logstreams - Close stream processor
13:49:48.250 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
13:49:48.294 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream-1
13:49:48.294 [stream-1-write-buffer] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
13:49:48.294 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream-1 close 2 readers
13:49:48.294 [] DEBUG io.camunda.zeebe.broker.test - Clean up test files on path /tmp/junit14901977728875490396
13:49:48.295 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
13:49:48.295 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
13:49:48.296 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
13:49:48.296 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully

java.lang.AssertionError: 
Expecting actual:
  [2251799813685249L, 2251799813685252L, 2251799813685249L, 2251799813685252L]
to contain exactly (and in same order):
  [2251799813685249L,
    2251799813685252L,
    2251799813685249L,
    2251799813685252L,
    2251799813685252L]
but could not find the following elements:
  [2251799813685252L]


	at io.camunda.zeebe.engine.processing.timer.TimerStartEventTest.shouldTriggerOnlyTwice(TimerStartEventTest.java:1107)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)

```
</details>

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
@@ -1096,20 +1096,27 @@ public final class TimerStartEventTest {
     // when
     engine.increaseTime(Duration.ofSeconds(5));
 
+    // disable because we'll await with Awaitility
+    RecordingExporter.disableAwaitingIncomingRecords();
+
     // then
-    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(4))
-        .extracting(record -> record.getValue().getProcessDefinitionKey())
-        .containsExactly(
-            firstDeploymentProcessDefinitionKey,
-            secondDeploymentProcessDefinitionKey,
-            firstDeploymentProcessDefinitionKey,
-            secondDeploymentProcessDefinitionKey);
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              // due timers are only checked if the engine is not currently processing #10112
+              // so we may need to move the clock slightly for each check
+              engine.increaseTime(Duration.ofMillis(100));
+              assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(4))
+                  .extracting(record -> record.getValue().getProcessDefinitionKey())
+                  .containsExactly(
+                      firstDeploymentProcessDefinitionKey,
+                      secondDeploymentProcessDefinitionKey,
+                      firstDeploymentProcessDefinitionKey,
+                      secondDeploymentProcessDefinitionKey);
+            });
 
     // when
     engine.increaseTime(Duration.ofSeconds(10));
-
-    // disable because we'll await with Awaitility
-    RecordingExporter.disableAwaitingIncomingRecords();
 
     // then
     Awaitility.await()

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
@@ -1108,6 +1108,9 @@ public final class TimerStartEventTest {
     // when
     engine.increaseTime(Duration.ofSeconds(10));
 
+    // disable because we'll await with Awaitility
+    RecordingExporter.disableAwaitingIncomingRecords();
+
     // then
     Awaitility.await()
         .untilAsserted(

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
@@ -30,6 +30,7 @@ import java.time.LocalTime;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
 import java.util.stream.Collectors;
+import org.awaitility.Awaitility;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
@@ -1103,6 +1104,9 @@ public final class TimerStartEventTest {
             secondDeploymentProcessDefinitionKey,
             firstDeploymentProcessDefinitionKey,
             secondDeploymentProcessDefinitionKey);
+
+    // due timers are only checked if the engine is not currently processing #10112
+    Awaitility.await("until the engine is idle again").until(engine::hasReachedEnd);
 
     // when
     engine.increaseTime(Duration.ofSeconds(10));

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
@@ -1105,21 +1105,26 @@ public final class TimerStartEventTest {
             firstDeploymentProcessDefinitionKey,
             secondDeploymentProcessDefinitionKey);
 
-    // due timers are only checked if the engine is not currently processing #10112
-    Awaitility.await("until the engine is idle again").until(engine::hasReachedEnd);
-
     // when
     engine.increaseTime(Duration.ofSeconds(10));
 
     // then
-    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(5))
-        .extracting(record -> record.getValue().getProcessDefinitionKey())
-        .containsExactly(
-            firstDeploymentProcessDefinitionKey,
-            secondDeploymentProcessDefinitionKey,
-            firstDeploymentProcessDefinitionKey,
-            secondDeploymentProcessDefinitionKey,
-            secondDeploymentProcessDefinitionKey);
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              // due timers are only checked if the engine is not currently processing #10112
+              // so we may need to move the clock slightly for each check
+              engine.increaseTime(Duration.ofMillis(100));
+              assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(5))
+                  .describedAs("Expect that start_1 triggered twice and start_2 triggered thrice")
+                  .extracting(record -> record.getValue().getProcessDefinitionKey())
+                  .containsExactly(
+                      firstDeploymentProcessDefinitionKey,
+                      secondDeploymentProcessDefinitionKey,
+                      firstDeploymentProcessDefinitionKey,
+                      secondDeploymentProcessDefinitionKey,
+                      secondDeploymentProcessDefinitionKey);
+            });
   }
 
   @Test

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
@@ -30,6 +30,7 @@ import java.time.LocalTime;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
 import java.util.stream.Collectors;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 
@@ -88,6 +89,11 @@ public final class TimerStartEventTest {
     final ProcessBuilder builder = Bpmn.createExecutableProcess("process_4");
     builder.startEvent("start_4").timerWithCycle("R/PT2S").endEvent("end_4");
     return builder.startEvent("start_4_2").timerWithCycle("R/PT3S").endEvent("end_4_2").done();
+  }
+
+  @Before
+  public void controlTheClock() {
+    engine.getClock().pinCurrentTime();
   }
 
   @Test

--- a/test-util/src/main/java/io/camunda/zeebe/test/util/record/RecordingExporter.java
+++ b/test-util/src/main/java/io/camunda/zeebe/test/util/record/RecordingExporter.java
@@ -73,6 +73,18 @@ public final class RecordingExporter implements Exporter {
     RecordingExporter.maximumWaitTime = maximumWaitTime;
   }
 
+  /**
+   * Disables the default awaiting behavior of the recording exporter by setting the wait time to 0.
+   *
+   * <p>By default, the RecordingExporter awaits incoming records until all expected records are
+   * retrieved (the expectation is controlled with limit), or a maximumWaitTime is surpassed. The
+   * wait time can also be controlled with {@link #setMaximumWaitTime(long)} and is reset along with
+   * the recorded records by {@link #reset()}.
+   */
+  public static void disableAwaitingIncomingRecords() {
+    setMaximumWaitTime(0);
+  }
+
   @Override
   public void open(final Controller controller) {
     this.controller = controller;
@@ -110,7 +122,7 @@ public final class RecordingExporter implements Exporter {
   protected static <T extends RecordValue> Stream<Record<T>> records(
       final ValueType valueType, final Class<T> valueClass) {
     final Spliterator<Record<?>> spliterator =
-        Spliterators.spliteratorUnknownSize(new RecordIterator(), Spliterator.ORDERED);
+        Spliterators.spliteratorUnknownSize(new AwaitingRecordIterator(), Spliterator.ORDERED);
     return StreamSupport.stream(spliterator, false)
         .filter(r -> r.getValueType() == valueType)
         .map(r -> (Record<T>) r);
@@ -118,7 +130,7 @@ public final class RecordingExporter implements Exporter {
 
   public static RecordStream records() {
     final Spliterator<Record<? extends RecordValue>> spliterator =
-        Spliterators.spliteratorUnknownSize(new RecordIterator(), Spliterator.ORDERED);
+        Spliterators.spliteratorUnknownSize(new AwaitingRecordIterator(), Spliterator.ORDERED);
     return new RecordStream(
         StreamSupport.stream(spliterator, false).map(r -> (Record<RecordValue>) r));
   }
@@ -276,7 +288,7 @@ public final class RecordingExporter implements Exporter {
         records(ValueType.DECISION_EVALUATION, DecisionEvaluationRecordValue.class));
   }
 
-  public static class RecordIterator implements Iterator<Record<?>> {
+  public static class AwaitingRecordIterator implements Iterator<Record<?>> {
 
     private int nextIndex = 0;
 

